src/test/java/com/systelab/seed/patient/PatientResourceTest.java

```diff
@@ -67,10 +85,23 @@ private void testCreateInvalidPatient(Patient patient) {
     @Description("Create a Patient with invalid data")
     @Test
     public void testCreateInvalidPatient() {
-        testCreateInvalidPatient(getPatientData("", "Burrows", "jburrows@test.com"));
-        testCreateInvalidPatient(getPatientData("John", "", "jburrows@test.com"));
-        testCreateInvalidPatient(getPatientData("", "", "jburrows@test.com"));
-        testCreateInvalidPatient(getPatientData("John", "Burrows", "jburrows"));
+        testCreateInvalidPatient(getPatientData("", "", "", ""));
+        testCreateInvalidPatient(getPatientData("", "", "", "111"));
+        testCreateInvalidPatient(getPatientData("", "", "jburrows@example", ""));
+        testCreateInvalidPatient(getPatientData("", "", "jburrows@.com", "222"));
+        testCreateInvalidPatient(getPatientData("", "Jameson", "", ""));
+        testCreateInvalidPatient(getPatientData("", "Jameson", "", "111"));
+        testCreateInvalidPatient(getPatientData("", "Jameson", "@example.com", ""));
+        testCreateInvalidPatient(getPatientData("", "Jameson", "jj@.", "333"));
+        testCreateInvalidPatient(getPatientData("Thomas", "", "", ""));
+        testCreateInvalidPatient(getPatientData("Thomas", "", "", "111"));
+        testCreateInvalidPatient(getPatientData("Thomas", "", "jj@example.com", ""));
+        testCreateInvalidPatient(getPatientData("John", "", "jburrows@test,com", "222"));
+        testCreateInvalidPatient(getPatientData("Thomas", "Summers", "", ""));
+        testCreateInvalidPatient(getPatientData("Thomas", "Summers", "", "333"));
+        testCreateInvalidPatient(getPatientData("Thomas", "Summers", "tsms@example,com", ""));
+        testCreateInvalidPatient(getPatientData("John", "Burrows", "jburrows,", "345423"));
+        //TODO: All possible combinations and patterns(email)
```

Comment:
I also think that it's not required to test all possible combinations. We just need to test that each field is null at least once.
We need to add a step to test the email incorrect format.

Reply:
I agree. In an automated test we can add as cases as we want, but in this case, as a seed example, it should be better to left only one case of each mandatory field. Suggestion:
```suggestion
        testCreateInvalidPatient(getPatientData("", "", "", ""));
        testCreateInvalidPatient(getPatientData("", "Summers", "jsummers@email.com", "345423"));
        testCreateInvalidPatient(getPatientData("John", "", "jsummers@email.com", "345423"));
        testCreateInvalidPatient(getPatientData("John", "Summers", "", "345423"));
        testCreateInvalidPatient(getPatientData("John", "Summers", "jsummers@email.com", ""));
```

Reply:
To get a test clearer, the cases of email format wrong (and other invalid patient reasons), should be added in other steps, and the Action should indicate clearly what is the reason of the invalid data (the expected result only indicates the error code). For example:

@Description("Create a Patient with invalid data: mandatory fields empty: Name, Surname, DOB or email")
public void testCreatePatientWithMandatoryFieldsEmpty() {
    testCreateInvalidPatient...
}

@Description("Create a Patient with invalid data: email format")
public void testCreatePatientWithEmailFormatInvalid() {
    testCreateInvalidPatient...
}

@Description("Create a Patient with invalid data: unique fields: Medical Number")
public void testCreatePatientWithUniqueFieldsRepeated() {
    testCreateInvalidPatient...
}

Reply:
In the Patient entity only 'Name' and 'Surname' have the Non-Null constraint. Additionally, the email format is verified so I will include some tests with invalid emails.

The unique field is not defined in the Patient entity, so it's no point in verify that right now.

I think this two steps are more close to what is defined by the entity:
@Description("Create a Patient with invalid data: mandatory fields empty (name, surname)")
@Description("Create a Patient with invalid data: email format")